src/java_bytecode/java_bytecode_parser.cpp

```diff
@@ -333,19 +334,43 @@ void java_bytecode_parsert::get_class_refs()
     }
   }
 
-  for(const auto &m : parse_tree.parsed_class.fields)
+  for(const auto &field : parse_tree.parsed_class.fields)
   {
-    typet t=java_type_from_string(m.signature);
-    get_class_refs_rec(t);
+    typet field_type;
+    if(field.has_signature)
+    {
+      field_type=java_type_from_string(
+        field.signature,
+        "java::"+id2string(parse_tree.parsed_class.name));
+    }
+    else
+      field_type=java_type_from_string(field.descriptor);
+    get_class_refs_rec(field_type);
   }
 
-  for(const auto &m : parse_tree.parsed_class.methods)
+  for(const auto &method : parse_tree.parsed_class.methods)
   {
-    typet t=java_type_from_string(m.signature);
-    get_class_refs_rec(t);
-    for(const auto &var : m.local_variable_table)
+    typet method_type;
+    if(method.has_signature)
+    {
+      method_type=java_type_from_string(
+        method.signature,
+        "java::"+id2string(parse_tree.parsed_class.name));
+    }
+    else
+      method_type=java_type_from_string(method.descriptor);
```

Comment:
Can there be some spacing, or better yet, some curly braces enclosing the else statement to better denote intention?